contracts/HATKlerosV2Connector.sol

```diff
@@ -0,0 +1,140 @@
+        DismissResolution // Dismiss the claim.
+    }
+
+    struct DisputeStruct {
+        bytes32 claimId; // Id of the claim in HATVault contract.
+        uint256 externalDisputeId; // Id of the dispute created in Kleros court.
+        Decision ruling; // Ruling given by the arbitrator.
+        bool resolved; // True if the dispute has been resolved.
+    }
+
+    IArbitrator public immutable klerosArbitrator; // The kleros arbitrator contract (e.g. Kleros Court).
+    HATArbitrator public immutable hatArbitrator; // Address of the Hat arbitrator contract.
+    IHATVault public immutable vault; // Vault contract.
+    bytes public arbitratorExtraData; // Extra data for the arbitrator.
+
+    DisputeStruct[] public disputes; // Stores the disputes created in this contract.
+    mapping(bytes32 => bool) public claimChallenged; // True if the claim was challenged in this contract..
+    mapping(uint256 => uint256) public externalIDtoLocalID; // Maps external dispute ids to local dispute ids.
+
+    /** @dev Raised when a claim is challenged.
+     *  @param _claimId Id of the claim in Vault cotract.
+     */
+    event Challenged(bytes32 indexed _claimId);
+
+    /** @dev Constructor.
+     *  @param _klerosArbitrator The Kleros arbitrator of the contract.
+     *  @param _arbitratorExtraData Extra data for the arbitrator.
+     *  @param _hatArbitrator Address of the Hat arbitrator.
+     *  @param _vault Address of the vault.
+     *  @param _metaEvidence Metaevidence for the dispute.
+     */
+    constructor (
+        IArbitrator _klerosArbitrator,
+        bytes memory _arbitratorExtraData,
+        HATArbitrator _hatArbitrator,
+        IHATVault _vault,
+        string memory _metaEvidence
+    ) {
+        // TODO: add new IEvidence events once they're established.
+        //emit MetaEvidence(0, _metaEvidence);
+        
+        klerosArbitrator = _klerosArbitrator;
+        arbitratorExtraData = _arbitratorExtraData;
+        hatArbitrator = _hatArbitrator;
+        vault = _vault;
+    }
+
+    /** @dev Notify KlerosArbitrator that expert's committee decision was challenged. Can only be called by Hat arbitrator.
+     *  Requires the arbitration fees to be paid.
+     *  @param _claimId The Id of the active claim in Vault contract.
+     *  @param _evidence URI of the evidence to support the challenge.
+     *  Note that the validity of the claim should be checked by Hat arbitrator.
+     */
+    function notifyArbitrator(bytes32 _claimId, string calldata _evidence) external payable {
+        require(msg.sender == address(hatArbitrator), "Wrong caller");
+        require(!claimChallenged[_claimId], "Claim already challenged");
+
+        // Surplus amount will be used to get more votes. Regardless, the UI should demand the correct value.
+        uint256 arbitrationCost = getArbitrationCost();
+        require(msg.value >= arbitrationCost, "Should pay the full deposit.");
```

Comment:
If msg.vaule is bigger than arbitrationCost the extra funds should be refunded to the caller.

Reply:
Alright, but there is a slight problem that the caller is HatArbitrator in this case, and for the latter to do the refund it should do its own check of `msg.value` and arbitrationCost.

Reply:
Yeah that should be fine

Reply:
I pushed the change

Reply:
Thanks, but this only returns the amount not used, but doesn't move the extra funds back to the hat arbitrator to reimburse the caller no?

Reply:
You're correct, my bad. I pushed the fix